libraries/AP_Scripting/applets/Bendy_ruler.lua

```diff
@@ -0,0 +1,871 @@
+    LOW = 1,          -- Mild concern about this threat
+    HIGH = 2,         -- Craft is panicking and may take action to avoid
+    ENUM_END = 3      -- End of enum
+}
+-- MAV_COLLISION_SRC
+MAV_COLLISION_SRC = {
+    ADSB = 0,                         -- Source is ADSB_VEHICLE packets
+    MAVLINK_GPS_GLOBAL_INT = 1,        -- Source is MAVLink GPS_GLOBAL_INT
+    ENUM_END = 2                      -- End of enum
+}
+MAV_COLLISION_ACTION={
+    NONE=0, -- Ignore any potential collisions 
+    REPORT=1, -- Report potential collision 
+    ASCEND_OR_DESCEND=2, -- Ascend or Descend to avoid threat 
+    MOVE_HORIZONTALLY=3, -- Move horizontally to avoid threat 
+    MOVE_PERPENDICULAR=4, -- Aircraft to move perpendicular to the collision's velocity vector 
+    RTL=5, -- Aircraft to fly directly back to its launch point 
+    HOVER=6, -- Aircraft to stop in place 
+}
+
+
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+
+local gcs_threat = {
+    src = nil,  
+    src_id = 0,  
+    timestamp_ms = 0,  
+    _location = nil,  
+    _velocity = nil,  
+    threat_level = nil,  
+    closest_approach_xy = 0.0,  
+    closest_approach_z = 0.0,  
+    time_to_closest_approach = 0.0,  
+    distance_to_closest_approach = 0.0  
+}
+
+
+local avoidance_request = {
+    current_loc = nil,
+    target_loc = nil,
+    groundspeed = 0.0,
+    airspeed = 0.0,
+    wind_dir_rad = 0.0,
+    wind_speed = 0.0,
+    request_time_ms = 0
+}
+local avoid_req2 = {
+    current_loc = nil,
+    target_loc = nil,
+    groundspeed = 0.0,
+    airspeed = 0.0,
+    wind_dir_rad = 0.0,
+    wind_speed = 0.0,
+    request_time_ms = 0
+}
+local avoidance_result = {
+    target_loc = nil,
+    new_target_loc = nil,
+    result_time_ms = 0,
+    avoidance_needed = nil
+}
+
+local PARAM_TABLE_KEY = 7
+local PARAM_TABLE_PREFIX = "BENDY_"
+local OPTION_IGNORE_HEIGHT=1
+local ALT_FRAME_ABSOLUTE = 0
+
+-- bind a parameter to a variable
+function bind_param(name)
+   local p = Parameter()
+   assert(p:init(name), string.format('could not find %s parameter', name))
+   return p
+end
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+   assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', name))
+   return bind_param(PARAM_TABLE_PREFIX .. name)
+end
+
+-- setup SHIP specific parameters
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 7), 'could not add param table')
+--[[
+  // @Param: BENDY_OPTIONS
+  // @DisplayName: ADS-B avoidance options
+  // @Description: Bitmask of behaviour options
+  // @Values: 0:Disabled,1:Enabled
+  // @User: Advanced
+--]]
+BENDY_OPTIONS     = bind_add_param('OPTIONS', 1, 1)
+
+--[[
+  // @Param: BENDY_MARGIN_FENCE
+  // @DisplayName: fence margin
+  // @Description: Avoidance margin for fence
+  // @User: Advanced
+--]]
+BENDY_MARGIN_FENCE = bind_add_param('MARGIN_FENCE', 2, 50)
```

Comment:
How about MRG - jus thinking we might need the characters later so
MRG_FEN
MRG_DYN
MRG_EXC